core/parser.cpp

```diff
@@ -52,6 +52,12 @@ std::string jsonnet_unparse_number(double v)
     return ss.str();
 }
 
+/** Maximum parsing depth to avoid stack overflow due to pathological or malicious code.
+ * This is especially important when parsing deeply nested structures that could lead to
+ * excessive recursion in the parser functions.
+ */
+static const unsigned MAX_PARSER_DEPTH = 1000;
```

Comment:
Nit: Move this into the anonymous namespace, to be consistent with the `EMPTY_FODDER` definition.